pygfx/renderers/wgpu/meshshader.py

```diff
@@ -329,21 +359,24 @@ def code_fragment(self):
 
             // Lighting
             $$ if lighting
-                let world_pos = varyings.world_pos;
+                // Get view direction
                 let view = select(
-                    normalize(u_stdinfo.cam_transform_inv[3].xyz - world_pos),
+                    normalize(u_stdinfo.cam_transform_inv[3].xyz - varyings.world_pos),
                     ( u_stdinfo.cam_transform_inv * vec4<f32>(0.0, 0.0, 1.0, 0.0) ).xyz,
                     is_orthographic()
                 );
-                // Get normal
-                var normal = vec3<f32>(varyings.normal);
-                $$ if flat_shading
-                let u = dpdx(varyings.world_pos);
-                let v = dpdy(varyings.world_pos);
-                normal = normalize(cross(u, v));
-                normal = select(normal, -normal, (select(0, 1, is_front) + u_stdinfo.flipped_winding) == 1);
+                // Get normal used to calculate lighting
+                $$ if use_normal_map is defined
+                    // Get normal from the normalmap and use the PBR logic to deal with back faces
+                    let normal_map = textureSample( t_normal_map, s_normal_map, varyings.texcoord ) * 2.0 - 1.0;
+                    let normal_map_scale = vec3<f32>( normal_map.xy * u_material.normal_scale, normal_map.z );
+                    let normal = perturbNormal2Arb(view, surface_normal, normal_map_scale, varyings.texcoord, is_front);
+                $$ else
+                    // Flip the surface normal if we're looking at the back face
+                    let normal = select(-surface_normal, surface_normal, is_front);
```

Comment:
After adding `$$ else` above, there should be no need to flip the normals again.
just `let normal = surface_normal`